examples/node.py

```diff
@@ -0,0 +1,303 @@
+import socket
+import sys
+import time
+import threading
+import random
+import hashlib
+
+from p2pnetwork.nodeconnection import NodeConnection
+
+"""
+Author: Maurice Snoeren <macsnoeren(at)gmail.com>
+Version: 0.3 beta (use at your own risk)
+Date: 7-5-2020
+
+Python package p2pnet for implementing decentralized peer-to-peer network applications
+
+TODO: Variabele to limit the number of connected nodes.
+TODO: Also create events when things go wrong, like a connection with a node has failed.
+"""
+
+class Node(threading.Thread):
+    """Implements a node that is able to connect to other nodes and is able to accept connections from other nodes.
+    After instantiation, the node creates a TCP/IP server with the given port.
+
+    Create instance of a Node. If you want to implement the Node functionality with a callback, you should 
+    provide a callback method. It is preferred to implement a new node by extending this Node class. 
+      host: The host name or ip address that is used to bind the TCP/IP server to.
+      port: The port number that is used to bind the TCP/IP server to.
+      callback: (optional) The callback that is invokes when events happen inside the network
+               def node_callback(event, main_node, connected_node, data):
+                 event: The event string that has happened.
+                 main_node: The main node that is running all the connections with the other nodes.
+                 connected_node: Which connected node caused the event.
+                 data: The data that is send by the connected node."""
+
+    def __init__(self, host, port, id, callback=None):
```

Comment:
When the id is not used, we could fall back on the automatic generation of the id as we had. Is that okay?